test/LibraryManager.IntegrationTest/AddCientSideLibrariesFromUITests.cs

```diff
@@ -0,0 +1,115 @@
+            _initialLibmanFileContent = File.ReadAllText(_pathToLibmanFile);
+
+            if (File.Exists(_libManConfig.FullPath))
+            {
+                string projectPath = Path.Combine(SolutionRootPath, _projectName);
+                _libManConfig.Delete();
+                Helpers.FileIO.WaitForDeletedFile(projectPath, Path.Combine(projectPath, _libman), caseInsensitive: false);
+            }
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromProjectRoot_SmokeTest()
+        {
+            SetLibraryAndClickInstall(_projectName, "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/lib/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromFolder_SmokeTest()
+        {
+            SetLibraryAndClickInstall("wwwroot", "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        private void SetLibraryAndClickInstall(string nodeName, string library)
+        {
+            SolutionExplorer.Select(nodeName);
+            InstallDialogTestService installDialogTestService = VisualStudio.Get<InstallDialogTestService>();
+            InstallDialogTestExtension installDialogTestExtenstion = installDialogTestService.OpenDialog();
+
+            installDialogTestExtenstion.SetLibrary(library);
+            installDialogTestExtenstion.WaitForFileSelections();
+            installDialogTestExtenstion.ClickInstall();
+        }
+
+        [TestCleanup]
```

Comment:
>TestCleanup [](start = 9, length = 11)

Maybe override DoHostTestCleanup?

Reply:
Updated in iteration 2

---
In reply to: [205457764](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205457764) [](ancestors = 205457764)